### PR TITLE
Wayland: Support integer scaling without wp_fractional_scale

### PR DESCRIPTION
**Pull request by bbb651**
Release Notes:
- N/A

`DoubleBuffered` is not currently very necessary because we only care about a single field `OutputState::scale` but I think it can be useful for other objects as it's a fairly common pattern in wayland.

### CODE REVIEW

**Review by mikayla-maki — CHANGES_REQUESTED** [review]

**Comment by bbb651** [comment]
I was also missing `wl_surface::commit`s after calls to `wl_surface::set_buffer_scale`

File: crates/gpui/src/platform/linux/wayland/client.rs [hunk 1]
```diff
@@ -324,6 +367,122 @@ impl Dispatch<WlCallback, Arc<WlSurface>> for WaylandClientState {
...
+        else {
+            return;
+        };
+
+        let mut outputs = window.outputs.borrow_mut();
+
+        match event {
+            wl_surface::Event::Enter { output } => {
+                // We use `PreferredBufferScale` instead to set the scale if it's available
+                if surface.version() >= 6 {
```
[the remark this answers is not in the document]
Comment (mikayla-maki) [reply crates/gpui/src/platform/linux/wayland/client.rs:401]: Yeah, write that down in a comment with the link!

**Comment by bbb651** [comment]
Turns out there are `wl_*::REQ_*_SINCE` constants already in `wayland-client`

**Comment by bbb651** [comment]
Sorry for all the CI errors and messy commits, I think it should be good to go now :smile:

**Review by mikayla-maki — APPROVED** [review]
lgtm!

**Pull request merged**